### ISSUE TITLE
Mouse movement not working

### ISSUE THREAD

**Issue by ETLegacyTeam** [etlegacy/etlegacy#774]
Author: Lammert (Lammert)
Date: 2015-01-11
Redmine Issue: 774, https://dev.etlegacy.com/issues/774

---

In some cases it seems that the mouse movement is not working. 
This only happens ingame (as spectator aswell as player). In menu it is working. 
Mouse click works when this problem appears. 

Already found out that restarting sometimes solves the issue, but not always.

Also there is a report that ALT+TAB solved the issue.

---

Related commit(s):
d30f866253b4efe0a826a927763a8877f69ccff9 - client: set the windows process affinity on startu..
 dcf19e86b653df21eb3c62f942a7b9bc3ade5808 - client: force at least 1 frame_msec value, refs 7..
 6bfdd80d62c02e6bf1cc3d7954874e040110f677 - general: disabled win32 cpu affinity code, fixes ..
 c803e0274cbaa124d68192a6d03608ab0d2a0f3f - properly disabled the affinity code, refs 774

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2015-01-11T13:30:39Z

---

Responsible commit seems to be "d098af23d187f016d8efc752e33e0cad2f752c08":https://github.com/etlegacy/etlegacy/commit/d098af23d187f016d8efc752e33e0cad2f752c08.

(Don't take it as granted, as the issue is quite random and I might have misbisected).

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2015-01-11T21:16:26Z

---

Setting CPU affinity to the last core only (CPU3 on my quadcore) seems to solve the issue.

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2015-01-12T11:52:33Z

---

Some helpful link from OJK (thx Ensi!):

https://github.com/JACoders/OpenJK/blob/master/codemp/win32/win_shared.cpp#L261
https://github.com/JACoders/OpenJK/blob/master/codemp/sys/sys_unix.cpp#L648
https://github.com/JACoders/OpenJK/blob/master/codemp/qcommon/common.cpp#L1263

This code set com_affinity  on the first core by default and that func is called from qcommon/common.cpp.

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2015-02-16T10:33:30Z

---

So I have a _good_ and a _bad_ news.

The good news is that bug doesn't seem directly related to the code. I was having that bug constantly, with the mouse being stuck after about a minutes of play. Since about ten days ago (5-6 of February), I cannot reproduce this problem. It went from "being always there" to "completely fixed". I tried to bisect the issue without success. I even tried the 2.72 official release (was buggy for me, although it was a bit better than 2.71a) and it works as expected. The fix seems somewhat unrelated to the code.

The bad news is, of course, that I have no single idea why it works now. As far as I remember, when it was buggy I tried to launch etl.exe specifically on "1 single core with a shortcut":http://www.eightforums.com/tutorials/40339-cpu-affinity-shortcut-program-create-windows.html (with affinity 1), which didn't really worked. After that, launching etl.exe directly (not through shortcut) was still buggy... and then it suddenly worked the next few days.

tldr: fixed on ma machine, but I don't know why :(

**Comment by ETLegacyTeam** [comment]
From: Mateos (Mateos)
Date: 2015-02-16T14:49:30Z

---

Under Windows with a Microsoft mouse ? May be a Microsoft Mouse & Keyboard Center update?

Tried to reset the CFG file and try with a new one?

Maybe the OS was hanging and selected to disconnect temporarily the mouse? (happens to me from time to time)

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2015-02-16T15:00:08Z

---

Mateos wrote:
> Under Windows with a Mircosoft mouse ? May be a Microsoft Mouse & Keyboard Center update?
> Tried to reset the CFG file and try with a new one?

I've always used a new config file. This shouldn't be hardware dependent, since it happens with my integrated touchpad (no mouse!) and that I haven't done any drivers update recently.
 
> Maybe the OS was hanging and selected to disconnect temporarily the mouse? (happens to me from time to time)

Maybe.... what could cause this issue?

**Comment by ETLegacyTeam** [comment]
From: @Julio (The_Loko)
Date: 2015-02-20T21:56:45Z

---

I still have this bug on Linux

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2015-02-20T22:21:18Z

---

The_Loko wrote:
> I still have this bug on Linux

Could give details about your issue? You're the first person to have it under Linux, so I suspect this is another bug..

**Comment by ETLegacyTeam** [comment]
From: @Julio (The_Loko)
Date: 2015-02-21T16:14:29Z

---

Spyhawk wrote:
> The_Loko wrote:
> > I still have this bug on Linux
> 
> Could give details about your issue? You're the first person to have it under Linux, so I suspect this is another bug..

I think it's the same, first time It happened was the day this bug was reported.
Most of times, when connecting to a server mouse movement doesn't work. Alt Tab doesn't fix it, restarting the game does.
As this one, mouse click works and movement works on menu too.

Also, after playing in a server a lot of maps (around 10), mouse stops working too when loading a new map.

**Comment by ETLegacyTeam** [comment]
From: @Julio (The_Loko)
Date: 2015-03-05T15:16:39Z

---

I tried 2.72 on my Windows 7 machine and I have the same issue.

**Comment by ETLegacyTeam** [comment]
From: Anonymous (Anonymous)
Date: 2015-03-06T13:51:31Z

---

Hello,

I am following this issue since the day it was first reported, because I have the same issue too, on Win 8 64-bit.
Some thoughts:
The issue only occures when I update to 2.72. With 2.71a the bug is not there. Maybe I am wrong but the bug must happen somewhere between 2.71a and 2.72.
There are not too much changes between 2.71a and 2.72. Can anyone who can reproduce the bug test 2.71a exe instead of 2.72 exe?
I can't compile ETLegacy, but if someone could, than it might not be to hard to exactly find out where the bug first occurs (around 24/12/2014 to 08/01/2015, maybe: https://github.com/etlegacy/etlegacy/commit/857efa2ab65adb701906071797561149fdcc67b8)
Hmm, I have no idea if my assumptions are correct, I simply thought maybe this information will help you a bit.

Kind Regards,
ToKu

**Comment by ETLegacyTeam** [comment]
From: @saukko28 (Saukko)
Date: 2015-03-06T15:10:15Z

---

I have been wondering why there haven't been any investigations after the problem occured.

-*S

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2015-03-06T15:43:20Z

---

ToKu> The exact commit is known (see #774#note-2 above), but it is still unclear why the bug happens, and why it can suddenly disappear on the very same machine with the very same code (see #774#note-9).

**Comment by ETLegacyTeam** [comment]
From: pisek (pisek)
Date: 2015-03-09T10:26:21Z

---

Spyhawk wrote:
> ToKu> The exact commit is known (see #774#note-2 above), but it is still unclear why the bug happens, and why it can suddenly disappear on the very same machine with the very same code (see #774#note-9).

In my case the problem shows up only when running the game with the omni-bots mode. In vanilla etlegacy the mouse works OK.

The mouse works at first OK, but after a minute of play, stops. The problem however is not with the mouse itself because when ESC is pressed, the cursor can be moved; when going back to game the mouse stops...
When I "disconnect" from the server and connect/create it again, it goes again (1min ok, then bump)

I am using:
Windows 7 64bit (clean system)
ET Legacy 2.72

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2015-03-09T14:35:24Z

---

> In my case the problem shows up only when running the game with the omni-bots mode.

Very interesting. I can't reproduce the issue anymore, but I had the exact same issue as you. could you tell us a bit more about this issue? Were you using a local server, or an online server? Did you use the "launch ETL with bots" shortcut? etc.

**Comment by ETLegacyTeam** [comment]
From: Anonymous (Anonymous)
Date: 2015-03-09T16:47:49Z

---

@Spyhawk> I had a look at the code (#774#note-2 above). I can't see anything which seems wrong with that commit, thought I won't argue against you, you might know more than I, I'm not that skilled and I am no ET Legacy developer. So it is guaranteed that this commit causes the bug?

I can confirm what Pisek wrote: mouse movement will stop working with omnibots enabled (tested on local server, no shortcut).

@Pisek> I'm not sure but to me it seems it mostly happens right after 1st respawn after I was killed for the first time, can you confirm this?

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2015-03-09T16:58:56Z

---

ToKu wrote:
> @Spyhawk> I had a look at the code (#774#note-2 above). I can't see anything which seems wrong with that commit, thought I won't argue against you, you might know more than I, I'm not that skilled and I am no ET Legacy developer. So it is guaranteed that this commit causes the bug?

Yes, it is the commit where the bug appears. However, it doesn't mean that code is directly responsible for it. Likely, it made it appear by triggering something else in another part of the code. Hopefully I'll be able to reproduce it with playing with OB settings.

**Comment by ETLegacyTeam** [comment]
From: @IR4T4 (IR4T4)
Date: 2015-03-09T17:05:06Z

---

To all players getting this bug:

If this issue occures open console (if that works) and enter:

```
cvarlist
condump mousebug.txt

```

You'll find the mousebug.txt file in fs_homepath. Please attach this file here. Don't forget to  anonymise before (name, GUID etc)

**Comment by ETLegacyTeam** [comment]
From: pisek (pisek)
Date: 2015-03-09T17:23:14Z

---

Unfortunatelly i have it reproduced on non-bot game as well... so no hint here...
But... I have just changed the sensitivity property to some other value and saved the profile, restarted game and mouse worked - so then just changed the sensitivity back to normal.
If the error will happen again i will surely dump the logs:)

**Comment by ETLegacyTeam** [comment]
From: pisek (pisek)
Date: 2015-03-09T17:26:57Z

---

ToKu wrote:
> @Pisek> I'm not sure but to me it seems it mostly happens right after 1st respawn after I was killed for the first time, can you confirm this?

In my case the mouse problem was fully random: once just in the beginning as spectator, sometimes after 1min of playing etc so no rule here:/

**Comment by ETLegacyTeam** [comment]
From: belstgut (belstgut)
Date: 2015-03-11T20:30:12Z

---

condump
http://hastebin.com/umifuhiced.txt

**Comment by ETLegacyTeam** [comment]
From: @IR4T4 (IR4T4)
Date: 2015-03-15T10:15:24Z

---

belstgut, your condump on hastebin is gone :/ please attach it as text file.

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2015-03-15T14:42:19Z

---

I can finally reproduce the issue, on Windows only and with other mods than Legacy (reproduced on silent and nitmod so far).
This is not the exact same issue I was facing previously (mouse working but stopping about 1 minute later - as described by pisek above) but it is the one described by many other people (mouse not working at all on connect, and a few /reconnect can solve the issue). 

Unlike the previous mouse issue I had, it *doesn't show up in localhost*, but *does on online servers*. However, a new bisect showed that the "same commit":https://github.com/etlegacy/etlegacy/commit/d098af23d187f016d8efc752e33e0cad2f752c08 is also responsible for this issue, so it is certainly the same bug, or a variant of it.

com_minimized and com_unfocused were both set to 0 when this bug happened.

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2015-03-15T22:01:55Z

---

Possible fix pushed in "dcf19e86b653df21eb3c62f942a7b9bc3ade5808":https://github.com/etlegacy/etlegacy/commit/dcf19e86b653df21eb3c62f942a7b9bc3ade5808. Feedback welcome!

Thanks to Razish (QtZ) for the "fix":https://github.com/Razish/QtZ/commit/2cd83291050f6b4af35fcf5777b47cc310020302#diff-37ffbd42e72139cff4e4e8a74fc555c6R554 and Ensiform for the help!

Note: I'm not sure if the affinity change (d30f8662) is still necessary.

**Comment by ETLegacyTeam** [comment]
From: Zecklar (Zecklar)
Date: 2015-03-16T18:57:45Z

---

The fix is correct.

What happens is division by 0 in CL_Keystate. If frame_msec = 0, in CL_Keystate the line val = (float)msec / frame_sec calculates the result as #IND ( indeterminate ) and in consequence of that, the viewangles ( cl.viewangles ) will be also #IND and so player can't look around anymore.

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2015-03-17T13:26:17Z

---

Applied in changeset commit:6bfdd80d62c02e6bf1cc3d7954874e040110f677.

**Comment by ETLegacyTeam** [comment]
From: watchdog (watchdog)
Date: 2015-03-29T23:46:42Z

---

Hello,
my mouse didn't working.

How can i fix the problem

**Comment by ETLegacyTeam** [comment]
From: Mateos (Mateos)
Date: 2015-03-30T07:55:17Z

---

Either compile the source or wait for the next release :)

**Comment by ETLegacyTeam** [comment]
From: watchdog (watchdog)
Date: 2015-03-31T05:26:15Z

---

Mateos wrote:
> Either compile the source or wait for the next release :)

How can I compile the source?

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2015-03-31T12:58:58Z

---

See README file in the github repo. Also, next time please ask question on forums instead of in the bug tracker.
